src/util/sighash.rs

```diff
@@ -126,8 +126,9 @@ pub enum SigHashType {
 /// Possible errors in computing the signature message
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub enum Error {
-    /// Should never happen since we are always encoding, thus we are avoiding wrap the IO error
-    IoError,
+    /// Could happen only by using `*_encode_signing_*` methods with custom writers, engines writers
+    /// like the ones used in methods `*_signature_hash` don't error
+    Io(io::ErrorKind),
```

Comment:
Is there any particular reason for only keeping the kind?

Reply:
Only this
 https://github.com/rust-bitcoin/rust-bitcoin/pull/628#discussion_r672234789

Reply:
Oh right, totally missed that, good point!